framework/src/meshgenerators/XYDelaunayGenerator.C

```diff
@@ -228,18 +232,28 @@ XYDelaunayGenerator::generate()
   std::vector<TriangulatorInterface::MeshedHole> meshed_holes;
   std::vector<TriangulatorInterface::Hole *> triangulator_hole_ptrs(_hole_ptrs.size());
   std::vector<std::unique_ptr<MeshBase>> hole_ptrs(_hole_ptrs.size());
+  // This tells us the element orders of the hole meshes
+  // For the boundary meshes, it can be access through poly2tri.segment_midpoints.
+  std::vector<bool> holes_with_midpoints(_hole_ptrs.size());
+  bool stitch_second_order_holes(false);
 
   // Make sure pointers here aren't invalidated by a resize
   meshed_holes.reserve(_hole_ptrs.size());
   for (auto hole_i : index_range(_hole_ptrs))
   {
     hole_ptrs[hole_i] = std::move(*_hole_ptrs[hole_i]);
     meshed_holes.emplace_back(*hole_ptrs[hole_i]);
+    holes_with_midpoints[hole_i] = meshed_holes.back().n_midpoints();
+    stitch_second_order_holes =
+        (holes_with_midpoints.back() && _stitch_holes[hole_i]) || stitch_second_order_holes;
     if (hole_i < _refine_holes.size())
       meshed_holes.back().set_refine_boundary_allowed(_refine_holes[hole_i]);
 
     triangulator_hole_ptrs[hole_i] = &meshed_holes.back();
   }
+  if (stitch_second_order_holes && (_tri_elem_type == "TRI3" || _tri_elem_type == "DEFAULT"))
+    paramError("tri_element_type",
+               "Cannot use first order elements with stitched quadratic element holes.");
 
```

Comment:
We're not trying to autodetect the order of the input `mesh`, right?  So if we try to stitch a quadratic boundary to quadratic holes, we'll scream and die here, even on the `DEFAULT` setting?  That's fine for now; the autodetection should go in on the libMesh side later, but in the meantime maybe we should add something like "Specify a higher-order tri_element_type or reduce the order of the hole inputs." to help users who hit this error?

Reply:
So far `DEFAULT` is just first order. This will change in the future once we have autodetection in the future.

I will add your suggestion to help users solve the issue.